### ISSUE TITLE
Debugger periodically crashes when fetching completions for debug console

### ISSUE THREAD

**Issue by weswigham** [microsoft/vscode-js-debug#1537]
**Describe the bug**
Debugger crashes when fetching completions, immediately terminating the debug session.

**To Reproduce**

I'm unsure how much, exactly, is required, but I've just been debugging the typescript compiler with node 16.13.0, and this has started to happen a _lot_ - this is just one way I've recorded where it seems to happen pretty regularly, but there may be a timing element to it.

Example steps to reproduce the behavior:
1. Clone https://github.com/microsoft/TypeScript
2. `npm ci` in the root
3. Copy `launch.template.json` to `launch.json` in the `.vscode` folder
4. Open `src/compiler/checker.ts` in one panel
5. Set a breakpoint on the first line of `inferFromTypes` (probably around line 23978) 
6. Open `tests\cases\conformance\types\typeParameters\typeArgumentLists\instantiationExpressions.ts` in another panel
7. Press f5 to launch the test with the debugger attached
8. Hit breakpoint
9. Go to the debug console and start typing out "originalSource.__debugTy" and press tab (completions probably aren't in yet, so this will do nothing)
10. Witness debug session abruptly end after a delay, see new error from the debugger in `Window` output log.

**Log File**

```
2023-01-30 13:09:45.937 [error] Error processing completions: Error: unreachable
	at d.provideForSource (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1048241)
	at c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1047966
	at runNextTicks (node:internal/process/task_queues:61:5)
	at process.processTimers (node:internal/timers:499:9)
	at async Q.getRename (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:975779)
	at async L.toDap (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:969837)
	at async Promise.all (index 0)
	at async n.getVariables (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:977957)
	at async c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:929019
	at async Promise.all (index 0)
	at async y.completions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:929122)
	at async f.defaultCompletions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:818538)
	at async f.identifierCompleter (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:816441)
	at async j.completions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:942065)
	at async t.default._onMessage (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1082470): CodeExpectedError: Error processing completions: Error: unreachable
	at d.provideForSource (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1048241)
	at c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1047966
	at runNextTicks (node:internal/process/task_queues:61:5)
	at process.processTimers (node:internal/timers:499:9)
	at async Q.getRename (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:975779)
	at async L.toDap (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:969837)
	at async Promise.all (index 0)
	at async n.getVariables (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:977957)
	at async c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:929019
	at async Promise.all (index 0)
	at async y.completions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:929122)
	at async f.defaultCompletions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:818538)
	at async f.identifierCompleter (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:816441)
	at async j.completions (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:942065)
	at async t.default._onMessage (c:\Users\wwigh\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\extension.js:2:1082470)
    at d.P (vscode-file://vscode-app/c:/Users/wwigh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1004:44287)
    at vscode-file://vscode-app/c:/Users/wwigh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1004:43781
    at async Object.provideCompletionItems (vscode-file://vscode-app/c:/Users/wwigh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:2235:4078)
    at async vscode-file://vscode-app/c:/Users/wwigh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:680:15687
    at async Promise.all (index 0)
    at async m (vscode-file://vscode-app/c:/Users/wwigh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/workbench.desktop.main.js:680:15559)
    at async Promise.all (index 0)
```

**VS Code Version:** 1.74.3

**Additional context**
I think it's [this throw](https://github.com/microsoft/vscode-js-debug/blob/4a1129e11a8b6d297fcbe2f56eec2d33273909ae/src/common/sourceMaps/renameProvider.ts#L78) the log is referencing. Evidently it's reachable :P

**Comment by connor4312** [comment]
Thanks for the issue. Have you hit this recently? I tried checking out the TS repo on a commit from the day you filed this issue and follow your repro instructions, but I was not successful in hitting the bug.

**Comment by mjbvz** [comment]
I'm seeing this all the time while trying to debug. @connor4312 How can I collect more detailed logs? Even in trace I don't see any logged errors

**Comment by connor4312** [comment]
Sharing a log is a good start. It might help me track down how it's getting thrown.

**Comment by mjbvz** [comment]
Here's the log: [vscode-debugadapter-6fba0048.json.gz](https://github.com/microsoft/vscode-js-debug/files/11205086/vscode-debugadapter-6fba0048.json.gz)

It seems to crash for me when I type `.` to access a property

**Comment by connor4312** [comment]
@mjbvz that is because tree-sitter is causing a segfault when accessing some properties. This can be easily reproduced with the following code:

```
const Parser = require('tree-sitter');
const TypeScript = require('./typescript');

const parser = new Parser();
parser.setLanguage(TypeScript);

const sourceCode = 'let x = 1; console.log(x);';
const tree = parser.parse(sourceCode);

const SegfaultHandler = require('segfault-handler');
SegfaultHandler.registerHandler('crash.log');

console.log('get type', tree.rootNode.__proto__.type);
console.log('the program should not have crashed');
```

@weswigham I wonder if you're hitting the same or a similar thing

**Comment by mjbvz** [comment]
@connor4312 Good find! Can we surface crashes like this in the UI? I don't recall seeing any error when it happened

**Comment by connor4312** [comment]
We print "process exited with code <whatever>" to the debug console. Being able to show the segfault would be nice, but the [segfault handler](https://github.com/ddopson/node-segfault-handler) module I used here is a native binding that I couldn't automatically include in the debug bootloader. It'd be nice if we got a CDP event for it, but from js-debug's perspective the process spontaneously disappears with no further communication.
